### PR TITLE
Use array for transition page cookie

### PR DESCRIPTION
**Pull request by yong-jie**
## Problem

Usage of an object coupled with LRU cache is might not be the best way to track the a user's visit history.

## Solution

As discussed, this is the approach using an ordered array. This PR uses the newly-introduced dependency injection IoC mechanism to inject the functions. Need @JasonChong96 help to check if I'm doing this correctly.

**Important testing note**:

If you have tested the transition page using the old code (json object), you might need to clear your cookies since this one relies on it being an array instead.

### CODE REVIEW

**Comment by JasonChong96** [comment]
IoC LGTM

File: src/server/util/transitionPage.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
+import _ from 'lodash'
+import { injectable } from 'inversify'
+
+/**
+ * Herein contains utility functions pertaining to
+ * the functionality of the transition page.
+ */
+
+const MAX_SIZE_BYTES = 2000
```
Comment (JasonChong96) [inline src/server/util/transitionPage.ts:9]: Would it be better to have this in config.ts?

File: src/server/util/transitionPage.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    shortUrl: string,
+  ): string[] {
+    if (!cookie) return [shortUrl]
+    if (cookie.includes(shortUrl)) {
+      return _.without(cookie, shortUrl).concat(shortUrl)
+    }
+    let newCookie = _.concat(cookie, shortUrl)
+    while (newCookie.toString().length > MAX_SIZE_BYTES) {
+      newCookie = _.drop(newCookie)
+    }
```
Comment (kylerwsm) [inline src/server/util/transitionPage.ts]: should we take into account of the overhead when converting to base64 later on over here?

File: src/server/util/transitionPage.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    shortUrl: string,
+  ): string[] {
+    if (!cookie) return [shortUrl]
+    if (cookie.includes(shortUrl)) {
+      return _.without(cookie, shortUrl).concat(shortUrl)
+    }
+    let newCookie = _.concat(cookie, shortUrl)
+    while (newCookie.toString().length > cookieSessionMaxSizeBytes) {
+      newCookie = _.drop(newCookie)
+    }
```
Comment (kylerwsm) [inline src/server/util/transitionPage.ts:55]: should we take into account of the base64 encoding overhead here?

File: src/server/util/transitionPage.ts [hunk 4]
```diff
@@ -47,7 +47,7 @@ export class CookieArrayReducer implements CookieReducer {
     shortUrl: string,
   ): string[] {
     if (!cookie) return [shortUrl]
-    if (this.userHasVisitedShortlink(cookie, shortUrl)) {
+    if (cookie.includes(shortUrl)) {
```
Comment (liangyuanruo) [inline src/server/util/transitionPage.ts:49]: what's the weird inversify bug?

Comment (yong-jie) [reply src/server/util/transitionPage.ts:49]: even though it's declared as a member of this class, when it is `bind`ed by inversify, `this` resolves to `undefined`

Comment (JasonChong96) [reply src/server/util/transitionPage.ts:49]: Using an arrow function avoids javascript's weird 'this' binding behavior

**Pull request merged**